lib/Doctrine/DBAL/Cache/QueryCacheProfile.php

```diff
@@ -43,9 +47,12 @@ public function __construct($lifetime = 0, ?string $cacheKey = null, ?object $re
         if ($resultCache instanceof Cache) {
             @trigger_error(sprintf('Using an instance of %s as result cache is deprecated. Please provide a PSR-6 cache instead.', Cache::class), E_USER_DEPRECATED);
 
-            $resultCache = new DoctrineCachePool($resultCache);
+            $resultCache = new DoctrineAdapter($resultCache);
```

Comment:
There is a small type mismatch here, since the Symfony adapter takes a `CacheProvider` instead of a `Cache`.